sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -561,15 +608,15 @@ private void acquireChannel(final Promise<Channel> promise) {
      */
     private void addTaskToPendingAcquisitionQueue(Promise<Channel> promise) {
 
-        this.ensureInEventLoop();
+//        this.ensureInEventLoop();
```

Comment:
Can these be made fully deterministic (like do either in the EventLoop or !EventLoop)?

Reply:
There is simply no need for this to run on the event loop's thread - but no harm in doing so either - actually for perf it is better to do this inline to eagerly pump pending acquisitions.